### PR TITLE
[Feature]: Add MFF

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpretrain/datasets/transforms/processing.py [hunk 1]
```diff
@@ -1740,3 +1744,52 @@ def transform(self, results: dict) -> dict:
...
+
+        w = min(w, width)
+        h = min(h, height)
+
+        i = torch.randint(0, height - h + 1, size=(1, )).item()
+        j = torch.randint(0, width - w + 1, size=(1, )).item()
+
+        return i, j, h, w
+
+    def forward(self, results: dict) -> dict:
```
Comment (fangyixiao18) [inline mmpretrain/datasets/transforms/processing.py:1780]:
```suggestion
    def transform(self, results: dict) -> dict:
```

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: mmpretrain/models/selfsup/__init__.py [hunk 2]
```diff
@@ -20,37 +21,10 @@
...
-    'DenseCL',
-    'BarlowTwins',
-    'SwAV',
-    'SparK',
+    'BaseSelfSupervisor', 'BEiTPretrainViT', 'VQKD', 'CAEPretrainViT',
+    'DALLEEncoder', 'MAEViT', 'MAEHiViT', 'iTPNHiViT', 'iTPN', 'HOGGenerator',
+    'MaskFeatViT', 'CLIPGenerator', 'MILANViT', 'MixMIMPretrainTransformer',
+    'MoCoV3ViT', 'SimMIMSwinTransformer', 'MoCo', 'MoCoV3', 'BYOL', 'SimCLR',
+    'SimSiam', 'BEiT', 'CAE', 'MAE', 'MaskFeat', 'MILAN', 'MixMIM', 'SimMIM',
+    'EVA', 'DenseCL', 'BarlowTwins', 'SwAV', 'SparK', 'MFF', 'MFFViT'
```
Comment (mzr1996) [inline mmpretrain/models/selfsup/__init__.py:29]: Please add the tail comma

File: mmpretrain/models/selfsup/mff.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            return (x, mask, ids_restore, proj_weights.view(-1))
+
+
+@MODELS.register_module()
+class MFF(MAE):
+    """MAE.
+
+    Implementation of `Masked Autoencoders Are Scalable Vision Learners
+    <https://arxiv.org/abs/2111.06377>`_.
+    """
```
Comment (mzr1996) [inline mmpretrain/models/selfsup/mff.py:170]: Please update the docstring.

File: mmpretrain/models/selfsup/mff.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        # to recover the original order of tokens in decoder.
+        latent, mask, ids_restore, weights = self.backbone(inputs)
+        pred = self.neck(latent, ids_restore)
+        loss = self.head.loss(pred, inputs, mask)
+        weight_params = {
+            f'weight_{i}': weights[i]
+            for i in range(weights.size(0))
+        }
+        losses = dict(loss=loss)
+        losses.update(weight_params)
```
Comment (mzr1996) [inline mmpretrain/models/selfsup/mff.py:194]: Why do you need to add these to the loss?

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: mmpretrain/datasets/transforms/processing.py (hunk 1 above)
Comment (YuanLiuuuuuu) [reply mmpretrain/datasets/transforms/processing.py:1780]: This class is the torchvision implementation, which should contain the ``forward`` func instead of ``transform``

File: mmpretrain/models/selfsup/mff.py (hunk 4 above)
Comment (YuanLiuuuuuu) [reply mmpretrain/models/selfsup/mff.py:194]: I want to log the weights dynamically during training, so I add them to the returned loss.

File: configs/mff/benchmarks/vit-base-p16_8xb128-coslr-100e_in1k.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    type='ImageClassifier',
+    backbone=dict(
+        type='VisionTransformer',
+        arch='base',
+        img_size=224,
+        patch_size=16,
+        drop_path_rate=0.1,
+        out_type='avg_featmap',
+        final_norm=False,
+        init_cfg=dict(type='Pretrained', checkpoint='')),
```
Comment (Ezra-Yu) [inline configs/mff/benchmarks/vit-base-p16_8xb128-coslr-100e_in1k.py:60]: prefix='backbone'

**Review by Ezra-Yu — COMMENTED** [review]
Please add readme, metafile, and add the metafile path to `https://github.com/open-mmlab/mmpretrain/blob/main/model-index.yml`

**Review by fangyixiao18 — APPROVED** [review]
